staged-documentation.md

```diff
@@ -0,0 +1,282 @@
+
+```java
+import com.commercetools.sunrise.common.contexts.UserContext;
+import com.commercetools.sunrise.common.pages.PageData;
+import com.commercetools.sunrise.components.ComponentBean;
+import com.commercetools.sunrise.framework.ControllerComponent;
+import com.commercetools.sunrise.hooks.PageDataHook;
+import io.sphere.sdk.models.Base;
+import play.Configuration;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class SummerCampaignControllerComponent extends Base implements ControllerComponent, PageDataHook {
+
+    @Inject
+    private Configuration configuration;
+    @Inject
+    private UserContext userContext;
+
+    @Override
+    public void acceptPageData(final PageData pageData) {
+        final ComponentBean component = createComponentBean();
+        pageData.getContent().addComponent(component);
+    }
+
+    private ComponentBean createComponentBean() {
+        final ComponentBean component = new ComponentBean();
+        //the template is in conf/templates/components/summercampaign/banner.hbs or of course your theme
+        component.setTemplateName("components/summercampaign/banner");//without .hbs!!!
+        Map<String, Object> data = new HashMap<>();
+        data.put("src", configuration.getString("summercampaign.img.src"));
+        data.put("width", configuration.getString("summercampaign.img.width"));
+        data.put("height", configuration.getString("summercampaign.img.height"));
+        data.put("href", demo.productcatalog.routes.SummerCampaignController.show(userContext.languageTag()).url());
+        component.setComponentData(data);
+        return component;
+    }
+}
+```
+
+You still need to wire the `ControllerComponent` to a controller or multiple controllers.
+Since it make sense to put it to all controllers except for checkout, we need a Guice Module:
+
+```java
+import com.commercetools.sunrise.framework.MultiControllerComponentResolver;
+import com.commercetools.sunrise.framework.MultiControllerComponentResolverBuilder;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import demo.productcatalog.SummerCampaignControllerComponent;
+
+public class ComponentsModule extends AbstractModule {
+    @Override
+    protected void configure() {
+    }
+
+    @Provides
+    public MultiControllerComponentResolver multiControllerComponentResolver() {
+        return new MultiControllerComponentResolverBuilder()
+                .add(SummerCampaignControllerComponent.class, controller -> !controller.getFrameworkTags().contains("checkout"))
+                .build();
+    }
+}
+```
+
+The `ComponentsModule` can be reused for multiple components, just add a `add` line for building the `MultiControllerComponentResolver`.
+
+Don't forget to register the module in `application.conf` with
+
+```
+play.modules.enabled += "absolute.path.to.your.ComponentsModule"
```

Comment:
Yes, that can be an ongoing discussion. Maybe it also makes sense to merge the modules for factories and components, because components may take part in using the overridden factories.